### PR TITLE
feat(gallery): add support for creating PhotoSwipe gallery from the I…

### PR DESCRIPTION
**Pull request by talha131**

### CODE REVIEW

File: documentation/content/Components/photoswipe-instagramp-gallery.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+Title: Gallery -- Use Instagram
+authors: Talha Mansoor, Pablo Iranzo Gómez
+Tags: nuances, images, gallery, instagram
+Date: 2020-02-06
+Slug: photoswipe-gallery-using-instagram
+Category: Components
+---
+
+Similar to the use case defined in [Photogallery]({filename}photogallery.md), Pelican has code in place for showing instragram galleries.
```
Comment (talha131) [inline documentation/content/Components/photoswipe-instagramp-gallery.md]: icorrect path to the article.

Comment (talha131) [inline documentation/content/Components/photoswipe-instagramp-gallery.md]: Capitalize Instagram

File: documentation/content/Components/photoswipe-instagramp-gallery.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+## Article contents
+
+To enable the plugin, just define a similar div inside your article for each gallery:
```
Comment (talha131) [inline documentation/content/Components/photoswipe-instagramp-gallery.md]: To embed Instagram gallery, just define a `div` in this manner,

File: documentation/content/Components/photoswipe-instagramp-gallery.md (hunk 1 above)
Comment (talha131) [reply documentation/content/Components/photoswipe-instagramp-gallery.md]: Incorrect spelling

File: documentation/content/Components/photoswipe-instagramp-gallery.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+---
+author: Pablo Iranzo Gómez
+title: Instagram
+tags: python, redken, descuenbot, deal, chollo, bargain, rabbate, soldes
+date: 2020-01-27 00:35:00 +0100
+comments: true
+category: blog
+description:
+lang: en
+---
```
Comment (talha131) [inline documentation/content/Components/photoswipe-instagramp-gallery.md]: This part is not required. Just tell the user about the `<div>`

File: documentation/content/Components/photoswipe-instagramp-gallery.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+Single image
+<div class="elegant-instagram" data-instagram-id="B7yh4IdItNd"></div>
+Multiple image
+<div class="elegant-instagram" data-instagram-id="BwWo35fAcR3"></div>
+```
+
```
Comment (talha131) [inline documentation/content/Components/photoswipe-instagramp-gallery.md]: Tell user class `elegant-instagram` is also important.

File: documentation/content/Components/photoswipe-instagramp-gallery.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+`https://www.instagram.com/p/OzF8OwS43q/` will mean adding to the div the parameter `data-instagram-id=OzF8OwS43q`.
+
+!!! hint "`data-instagram-id` defines either picture or set of pictures"
+
+    Note that `data-instagram-id` is the part in the url of a post, that can be a single or multiple pictures 'post' and in both case the library will show them.
+
+The above `article` code will then render as this:
+
+Single image
```
Comment (talha131) [inline documentation/content/Components/photoswipe-instagramp-gallery.md]: Use headings

File: documentation/content/Components/photoswipe-instagramp-gallery.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+<div class="elegant-instagram" data-instagram-id="B7yh4IdItNd"></div>
+Multiple image
```
Comment (talha131) [inline documentation/content/Components/photoswipe-instagramp-gallery.md]: Place multiple image example above the single image.

File: static/js/create-instagram-gallery.js [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+          if (
+            level1.edge_sidecar_to_children &&
+            level1.edge_sidecar_to_children.edges.length > 0
+          ) {
+            const username = level1.owner.username;
+            const name = level1.owner.full_name;
+            // It is more than one image
+            level1.edge_sidecar_to_children.edges.forEach(edge => {
+              console.log(edge);
```
Comment (talha131) [inline static/js/create-instagram-gallery.js]: Remove console.

File: documentation/content/Components/photoswipe-instagramp-gallery.md (hunk 1 above)
Comment (iranzo) [reply documentation/content/Components/photoswipe-instagramp-gallery.md]: fixed

File: documentation/content/Components/photoswipe-instagramp-gallery.md (hunk 2 above)
Comment (iranzo) [reply documentation/content/Components/photoswipe-instagramp-gallery.md]: done

File: documentation/content/Components/photoswipe-instagramp-gallery.md (hunk 1 above)
Comment (iranzo) [reply documentation/content/Components/photoswipe-instagramp-gallery.md]: Instagram capitalized

File: documentation/content/Components/photoswipe-instagramp-gallery.md (hunk 5 above)
Comment (iranzo) [reply documentation/content/Components/photoswipe-instagramp-gallery.md]: done

File: documentation/content/Components/photoswipe-instagramp-gallery.md (hunk 6 above)
Comment (iranzo) [reply documentation/content/Components/photoswipe-instagramp-gallery.md]: done

File: static/js/create-instagram-gallery.js (hunk 7 above)
Comment (iranzo) [reply static/js/create-instagram-gallery.js]: done

File: documentation/content/Components/photoswipe-instagramp-gallery.md (hunk 3 above)
Comment (iranzo) [reply documentation/content/Components/photoswipe-instagramp-gallery.md]: done

File: documentation/content/Components/photoswipe-instagramp-gallery.md (hunk 4 above)
Comment (iranzo) [reply documentation/content/Components/photoswipe-instagramp-gallery.md]: done

File: documentation/content/Components/photoswipe-instagramp-gallery.md (hunk 1 above)
Comment (iranzo) [reply documentation/content/Components/photoswipe-instagramp-gallery.md]: done

**Comment by iranzo** [comment]
> I have left comments for you. Also, the multiple image gallery is not working.

I've addressed the comments, it's curious as it works on my local system using gulp:

image

**Comment by iranzo** [comment]
Same on my  local 'make devserver' instance :

image

**Pull request merged**